core/src/main/resources/volcano-viz/viz-template.html

```diff
@@ -0,0 +1,323 @@
+        fill: #fff;
+        stroke-width: 1.5px;
+    }
+
+    .edgePath path {
+        stroke: #333;
+        stroke-width: 2px;
+    }
+
+    .container {
+        display: flex;
+        align-items: center;
+    }
+
+    .column1 {
+        flex: 0 0 300px;
+    }
+
+    .column2 {
+        flex: 0 0 1000px;
+    }
+
+    .tippy-content {
+        word-break: break-all;
+        word-wrap: break-word;
+    }
+</style>
+
+<div class="container">
+    <div class="column1">
+        <div style="width: 100%; text-align: center">
+            <div id="current-rule" style="display: block"></div>
+            <button id="prev-button" style="width: 80px; height: 40px; display:inline-block" disabled>prev </button>
+            <button id="next-button" style="width: 80px; height: 40px; display:inline-block" disabled> next</button>
+        </div>
+        <ul id="rule-match-list" style="width: 300px; height: 600px; overflow: auto">
+        </ul>
+    </div>
+    <div class="column2">
+        <svg id="svg-canvas" width="1200px" height="800px"></svg>
+    </div>
+</div>
+
+<script id="js">
+
+    var allNodes = data.allNodes;
+    var ruleMatchSequence = data.ruleMatchSequence;
+    var ruleMatchInfoMap = data.ruleMatchInfoMap;
+    var nodeAddedInRule = data.nodeAddedInRule;
+
+    /*
+     * Graph data and D3 JS render related variables
+     */
+
+    // Create the input graph
+    var g = new dagreD3.graphlib.Graph({
+            compound: true
+        })
+        .setGraph({
+            rankdir: 'LR'
+        })
+        .setDefaultEdgeLabel(function () {
+            return {};
+        });
+
+    // Create the renderer
+    var render = new dagreD3.render();
+
+    // Set up an SVG group so that we can translate the final graph.
+    var svg = d3.select("svg");
+    var svgGroup = svg.append("g");
+
+    // Set up zoom support
+    var svg = d3.select("svg")
+        .attr("width", "1200px")
+        .attr("height", "800px")
+        .call(d3.zoom().on("zoom", function () {
+            svgGroup.attr("transform", d3.event.transform)
+        }));
+
+    /*
+     * Global State
+     */
+
+    var currentRuleID = undefined;
+
+    /*
+     * Event Handler functions
+     */
+
+    var setCurrentRule = (ruleMatchID) => {
+        // un-highlight previous entry
+        var prevRuleID = currentRuleID;
+        if (prevRuleID !== undefined) {
+            var prevRuleElement = document.getElementById(prevRuleID);
+            prevRuleElement.style.backgroundColor = "#FFFFFF";
+        }
+
+        currentRuleID = ruleMatchID;
+        document.getElementById('current-rule').innerText = currentRuleID;
+
+        var currentRuleElement = document.getElementById(currentRuleID);
+        currentRuleElement.style.backgroundColor = "#D3D3D3";
+
+        var ruleIndex = ruleMatchSequence.indexOf(currentRuleID);
+
+        document.getElementById("prev-button").disabled = false;
+        document.getElementById("next-button").disabled = false;
```

Comment:
We can set the correct value directly (without using an if).

Reply:
agree